src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -2454,5 +2456,170 @@ public void SwitchExpression03()
+                                        N(SyntaxKind.IdentifierToken, "n");
+                                    }
+                                }
+                            }
+                            M(SyntaxKind.CloseParenToken);
+                        }
+                        M(SyntaxKind.ColonToken);
+                    }
+                    N(SyntaxKind.EmptyStatement);
+                    {
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.BreakStatement);
+                    {
+                        N(SyntaxKind.BreakKeyword);
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            EOF();
+        }
+
+        [Fact(Skip = "This is not a reliable test, and its failure modes are hard to capture. But it is helpful to run by hand to find parser issues.")]
+        public void ParseFuzz()
+        {
+            Random random = new Random();
+            string makeProps(int maxDepth, bool needNames)
+            {
+                int nProps = random.Next(maxDepth);
+                var builder = new StringBuilder();
+                for (int i = 0; i < nProps; i++)
+                {
+                    if (i != 0) builder.Append(", ");
+                    if (needNames || random.Next(5) == 0) builder.Append("N: ");
+                    builder.Append(makePattern(maxDepth - 1));
+                }
+                return builder.ToString();
+            }
+            string makePattern(int maxDepth)
+            {
+                if (maxDepth <= 0 || random.Next(6) == 0)
+                {
+                    switch (random.Next(4))
+                    {
+                        case 0:
+                            return "_";
+                        case 1:
+                            return "1";
+                        case 2:
+                            return "T x";
+                        default:
+                            return "var y";
+                    }
+                }
+                else
+                {
+                    // recursive pattern
+                    while (true)
+                    {
+                        bool nameType = random.Next(2) == 0;
+                        bool parensPart = random.Next(2) == 0;
+                        bool propsPart = random.Next(2) == 0;
+                        bool name = random.Next(2) == 0;
+                        if (!parensPart && !propsPart && !(nameType && name)) continue;
+                        return $"{(nameType ? "N" : "")} {(parensPart ? $"({makeProps(maxDepth, false)})" : "")} {(propsPart ? $"{{ {makeProps(maxDepth, true)} }}" : "")} {(name ? "n" : "")}";
+                    }
+                }
+            }
+            string makePattern0() => makePattern(random.Next(5));
+            for (int i = 0; i < 2000; i++)
```

Comment:
Nit: would you mind moving the local functions to the end of the test. It may be subjective, but I found this helps with following the flow of the method's core logic. Thanks